Polyfills/Console/Source/Console.cpp

```diff
@@ -34,16 +38,136 @@ namespace
         }
     }
 
+    bool EndsWith(std::string_view s, char c)
+    {
+        if (!s.empty())
+        {
+            return s.back() == c;
+        }
+        return false;
+    }
+
+    // based on: https://stackoverflow.com/a/26221725
+    template<typename T>
+    std::string FormatValue(std::string& formatString, T value)
+    {
+        // get the size of what will be written (+1 for null terminator)
+        int size_s = snprintf(nullptr, 0, formatString.c_str(), value) + 1;
+        // if size is 0 or less, we had a formatting error
+        if (size_s <= 0)
+        {
+            throw std::runtime_error("Error during console.log formatting.");
+        }
+        size_t size = static_cast<size_t>(size_s);
+        // create a string to hold that size (-1 as we don't want the null terminator)
+        std::string buf(size - 1, '0');
+        // write to there
+        snprintf(buf.data(), size, formatString.c_str(), value);
+        return buf;
+    }
+
     void InvokeCallback(Babylon::Polyfills::Console::CallbackT callback, const Napi::CallbackInfo& info, Babylon::Polyfills::Console::LogLevel logLevel)
     {
-        std::stringstream ss{};
-        for (size_t i = 0; i < info.Length(); i++)
+        std::ostringstream ss{};
+        if (info.Length() > 0)
         {
-            if (i > 0)
+            std::string firstArg = info[0].ToString();
+            size_t currArgIndex = 1;
+
+            // Split the first string into parts limited by substitution characters
+            size_t start = 0;
+            bool matching = false;
+            std::size_t j;
+            for (j = 0; j < firstArg.size(); j++)
+            {
+                const char currChar = firstArg.at(j);
+                if (!matching && currChar == '%')
+                {
+                    matching = true;
+                    if (j > 0)
+                    {
+                        std::string_view prefix(firstArg.data() + start, j - start);
+                        ss << prefix;
+                    }
```

Comment:
Would it work to add to `ss` when not matching instead of this? Also, I don't know what `prefix` means here.